### PR TITLE
Control Plane Taints: end-to-end tests; account for applied taints

### PR DESCRIPTION
**Pull request by danbudris**
*Issue #, if available:*
https://github.com/aws/eks-anywhere/issues/1226
https://github.com/aws/eks-anywhere/issues/189

*Description of changes:*
- Account for control plane taints in end-to-end tests
  - map a CP configuration back to a node, and compare their taints
  - if no taints are provided by the end-user, ensure the default taint is present
- Add methods to make validating node-to-configuration mappings much easier
  - add method `ValidateWorkerNodes` and `ValidateControlPlaneNodes`, which accept var arg validation functions
  - these methods then deduce the config-to-node mapping and run the provided validation functions, passing the node/config pair for each node
  - these can be later added to the internals of methods like `ValidateCluster` to call a variety of validation methods on the underlying nodes and abstract away the need to specifiy all of these validations at the test level. This refactoring is a future extension of this work.
- Update some existing configurations to ensure they tolerate custom CP taints
  - AWS IAM Auth now tolerate custom CP taints in addition to the default one
  - VSphere CSI drivers now tolerate custom CP taints in addition to the default one

As discussed in pervious PRs, unit tests for the new kubectl methods are coming in a separate PR to keep this one from getting too too out of hand.

*Testing (if applicable):*
Ran the taints tests on both BottleRocket and Ubuntu

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/framework/workerNodes.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+								e.T.Errorf("Worker node %v, member of Worker Node Group Configuration %v, is not valid: %v", node.Name, w.Name, err)
+							}
+						}
+					}
+				}
+			}
+		}
+		e.StopIfFailed()
+	}
+	e.T.Log("validated that expected taints are present on the workload cluster nodes")
```
Comment (danbudris) [inline test/framework/workerNodes.go:67]:
```suggestion
```

File: test/e2e/taints_test.go [hunk 2]
```diff
@@ -26,26 +34,35 @@ func runTaintsUpgradeFlow(test *framework.ClusterE2ETest, updateVersion v1alpha1
 // remove a taint from a node
 // add a taint to a node which already has another taint
 // add a taint to a node which had no taints
-func TestVSphereKubernetes121TaintsWorkerNodeGroups(t *testing.T) {
-	worker0 := "worker-0"
-	worker1 := "worker-1"
-	worker2 := "worker-2"
+func TestVSphereKubernetes121Taints(t *testing.T) {
```
Comment (vivek-koppuru) [inline test/e2e/taints_test.go:37]: Is the feature flag being removed with these changes? If not, you need to pass that in too right?

Comment (danbudris) [reply test/e2e/taints_test.go:37]: yes true, good call

Comment (danbudris) [reply test/e2e/taints_test.go:37]: done https://github.com/aws/eks-anywhere/pull/1254/commits/64727114bf2dbb536274b62fee8e4a91b08ef0d5

**Review by vivek-koppuru — APPROVED** [review]
/lgtm

**Pull request merged**